sdks/java/harness/src/test/java/org/apache/beam/fn/harness/logging/BeamFnLoggingClientTest.java

```diff
@@ -249,7 +251,7 @@ public StreamObserver<BeamFnApi.LogEntry.List> logging(
                       StreamObserver<BeamFnApi.LogControl> outboundObserver) {
                     // Block before returning an error on the stream so that we can observe the
                     // loggers before they are reset.
-                    streamBlocker.awaitAdvance(1);
```

Comment:
this should have been awaitAdvance(0) to have effect. Replaced with CompletableFuture which is simpler for this single block and then trigger.